### ISSUE TITLE
Resumability, caching, partial processing and interactivity

### ISSUE THREAD

**Issue by KOLANICH** [avast/retdec#875]
1. When a pass fails, it would be nice to be able to exclude it and continue. It requires retdec to store some state on disk, such as the plan of passes and auxilary info from each pass.
  1.a Currently even to redo a call to the decompiler to get both python-like and c-like pseudocodes after the lifting has been done one has to either restart the process from scratch. It is slow.
2. retdec often detects functions signatures and types wrong. For example some funcs with variable count of arguments are decompiled wrong, and the code near them looos meaningless too. It would be nice to have a way to fix such signatures and then to allow retdec propagate the changes. And this should be fast.

**Comment by BieHDC** [comment]
>     1. When a pass fails, it would be nice to be able to exclude it and continue. It requires retdec to store some state on disk, such as the plan of passes and auxilary info from each pass.
>        1.a Currently even to redo a call to the decompiler to get both python-like and c-like pseudocodes after the lifting has been done one has to either restart the process from scratch. It is slow.

Yes please. I currently have a job running that is already going for 2 weeks straight now, it would be really good if i could ctrl+c and later just continue where it left off.

**Comment by KOLANICH** [comment]
>going for 2 weeks straight now

Likely an endless loop. I have encounterd this on some binary.

**Comment by BieHDC** [comment]
> > going for 2 weeks straight now
> 
> There may be an endless loop within one of the passes. I have encountered hehavior looking like this on some pretty small binary. It worth to admit, that compared to hex-rays retdec currently sucks, I mean retdec cannot decompile (in my case the most frequently encountered failure mode is OOM) some binaries, that hex-rays had easily decompiled on the machines of several generations behind and with much less resource consumption.

No, it is still going, just some stages take a few hours, others a few days, even tho it uses 16gb ram and 28gb swap, the usage is not rising, but consistent. It is obviously the swapping that is choking the whole process.
